collector/infiniband.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+// +build !noinfiniband
+
+package collector
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+var noIBDevicesFound = errors.New("InfiniBand / No devices detected")
```

Comment:
I think they're still going to flag these, but if you want to keep them global, you will need to add some sort of prefix that uniquely identifies them. Probably change the naming structure to infinibandNoDevicesFound and infinibandNoPortsFound, based on what they told me with my last PR.